draft.go

```diff
@@ -0,0 +1,228 @@
+package grpc_prometheus
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/stats"
+)
+
+const (
+	namespace    = "grpc"
+	labelService = "grpc_service"
+	labelMethod  = "grpc_method"
+	//labelType      = "grpc_type"
+	//labelCode      = "grpc_code"
+	//labelUserAgent = "grpc_user_agent"
+	labelFailFast = "grpc_fail_fast"
+)
+
+type ctxKey int
+
+var (
+	tagRPCKey ctxKey = 1
+	//tagConnKey ctxKey = 2
+)
+
+type Subsystem int
+
+var (
+	Server Subsystem = 1
+	Client Subsystem = 2
```

Comment:
this seems unnecessary, why not do distinct functions/structs for server/client here?